cmake/Modules/DefineLocations.cmake.in

```diff
@@ -0,0 +1,12 @@
+# Define location paths and where to find stuff
```

Comment:
This is needed for the individual cmake scripts to gain knowledge about the tree structure of the project. The scripts are just another instance of cmake which does not inherit any of the variables defined in the instance that's running the build. In this manner, I am only passing it one variable which is the location of DefineLocations.cmake and the script will then know all the locations (eg. where to put generated header file or where to find the test executables).